packages/imperative/src/auth/handlers/BaseAuthHandler.ts

```diff
@@ -0,0 +1,233 @@
+                break;
+            default:
+                throw new ImperativeError({
+                    msg: `The group name "${commandParameters.positionals[1]}" was passed to the BaseAuthHandler, but it is not valid.`
+                });
+                break;
+        }
+    }
+
+    /**
+     * This is called by the {@link BaseAuthHandler#process} when it needs a
+     * session. Should be used to create a session to connect to the auth
+     * service.
+     * @abstract
+     * @param {ICommandArguments} args The command line arguments to use for building the session
+     * @returns {ISession} The session object built from the command line arguments.
+     */
+    protected abstract createSessCfgFromArgs(args: ICommandArguments): ISession;
+
+    /**
+     * This is called by the "auth login" command after it creates a session, to
+     * obtain a token that can be stored in a profile.
+     * @abstract
+     * @param {AbstractSession} session The session object to use to connect to the auth service
+     * @returns {Promise<string>} The response from the auth service containing a token
+     */
+    protected abstract async doLogin(session: AbstractSession): Promise<string>;
+
+    /**
+     * This is called by the "auth logout" command after it creates a session, to
+     * revoke a token before removing it from a profile.
+     * @abstract
+     * @param {AbstractSession} session The session object to use to connect to the auth service
+     */
+    protected abstract async doLogout(session: AbstractSession): Promise<void>;
+
+    /**
+     * Performs the login operation. Builds a session to connect to the auth
+     * service, sends a login request to it to obtain a token, and stores the
+     * resulting token in the profile of type `mProfileType`.
+     * @param {IHandlerParameters} params Command parameters sent by imperative.
+     */
+    private async processLogin(params: IHandlerParameters) {
+        const loadedProfile = params.profiles.getMeta(this.mProfileType, false);
+
+        const sessCfg: ISession = this.createSessCfgFromArgs(
+            params.arguments
+        );
+        const sessCfgWithCreds = await ConnectionPropsForSessCfg.addPropsOrPrompt<ISession>(
+            sessCfg, params.arguments,
+            { requestToken: true, defaultTokenType: this.mDefaultTokenType }
+        );
+
+        this.mSession = new Session(sessCfgWithCreds);
+
+        // login to obtain a token.
+        const tokenValue = await this.doLogin(this.mSession);
+
+        // update the profile given
+        if (loadedProfile != null && loadedProfile.name != null && !params.arguments.showToken && tokenValue != null) {
+            await Imperative.api.profileManager(this.mProfileType).update({
+                name: loadedProfile.name,
+                args: {
+                    "token-type": this.mSession.ISession.tokenType,
+                    "token-value": tokenValue
```

Comment:
Maybe I forgot, but why do we have to set the `hyphened-representation` and not the `camelCase` one?

Reply:
Because the option is specifically defined as `token-type`, and specifying the camel case one does not actually populate the option in the profile during the update because it does not go through Yargs.